core/src/services/gcs/writer.rs

```diff
@@ -32,11 +32,28 @@ pub struct GcsWriter {
 
     op: OpWrite,
     path: String,
+    upload_location: Option<String>,
+    already_uploaded_chunk: u64,
+    last_chunk_uploaded: bool,
```

Comment:
How about use `last: Option<Bytes>` instead?

Everytime users called `append`, we do the following thing:

- If last is none, store `bs` into last.
- If last is some, upload last to gcs, and replace with `bs`

While users call `close`:

- If last is none, users not call `append`, we can simply ignore this upload (I'm not sure about this).
- If last is some, upload this as last chunk.

This way, no extra data need to be resent.